plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/SharedHiveMetastoreCache.java

```diff
@@ -208,13 +203,10 @@ public HiveMetastore createMetastore(Optional<ConnectorIdentity> identity)
         private CachingHiveMetastore createUserCachingMetastore(String user)
         {
             ConnectorIdentity identity = ConnectorIdentity.ofUser(user);
-            return cachingHiveMetastore(
-                    metastoreFactory.createMetastore(Optional.of(identity)),
-                    new ReentrantBoundedExecutor(executorService, maxMetastoreRefreshThreads),
-                    metastoreCacheTtl,
-                    metastoreRefreshInterval,
-                    metastoreCacheMaximumSize,
-                    metastorePartitionCacheEnabled);
+            return cachingMetastoreBuilder
+                    .delegate(metastoreFactory.createMetastore(Optional.of(identity)))
```

Comment:
is `cachingMetastoreBuilder` a shared mutable state?

Reply:
Yes. I've made builder non-mutable